### PR TITLE
BAH-1237: Set CodedConceptDatatype from Id or Uuid

### PR DESCRIPTION
**Pull request by reagan-meant**

### CODE REVIEW

File: bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/customdatatype/datatype/CodedConceptDatatype.java [hunk 1]
```diff
@@ -1,19 +1,27 @@
...
     private Concept codedConcept;
 
     @Override
     public void setConfiguration(String id) {
-        this.codedConcept = Context.getConceptService().getConcept(Integer.valueOf(id));
+        if (MiscUtils.onlyDigits(id)) {
+            this.codedConcept = Context.getConceptService().getConcept(Integer.valueOf(id));
+        } else {
+            List<List<Object>> conceptId = Context.getAdministrationService()
+                    .executeSQL("select concept_id from concept where uuid='" + id + "';", true);
```
Comment (rbuisson) [inline bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/customdatatype/datatype/CodedConceptDatatype.java:22]: Is this pure SQL or could it be HQL instead?

**Review by rbuisson — CHANGES_REQUESTED** [review]

File: bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/customdatatype/datatype/CodedConceptDatatype.java (hunk 1 above)
Comment (rbuisson) [reply bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/customdatatype/datatype/CodedConceptDatatype.java:22]: Or, couldn't we use the `ConceptService`?

Comment (reagan-meant) [reply bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/customdatatype/datatype/CodedConceptDatatype.java:22]: @rbuisson For some unapparent reason the `conceptService` call by `uuid`and `name` throws an exception  ..Only the call by `conceptId` works.

Comment (rbuisson) [reply bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/customdatatype/datatype/CodedConceptDatatype.java:22]: Considering time constraints, we will merge this fix and come back to it later.

**Review by rbuisson — APPROVED** [review]

**Pull request merged**